applications/stepswitch/src/stepswitch_util.erl

```diff
@@ -91,61 +101,100 @@ correct_shortdial(<<"+", Number/binary>>, CIDNum, DeniedCallRestrictions) ->
                        ),
             'undefined';
         Length ->
-            try_to_correct(Number, CIDNum, DeniedCallRestrictions, Length)
-    end.
-
--spec try_to_correct(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:object(), non_neg_integer()) -> kz_term:api_ne_binary().
-try_to_correct(Number, CIDNum, DeniedCallRestrictions, Length) ->
+            maybe_deny_reclassified_number(do_correct_shortdial(Number, CIDNum, Length)
+                                          ,CIDNum
+                                          ,DeniedCallRestrictions
+                                          )
+    end;
+correct_shortdial(Number, _CIDNum, _DeniedCallRestrictions, {'false', Length}) ->
+    lager:debug("~s's lenght (~p) doesn't match min_shortdial_destination param's value (~p), skipping"
+               ,[Number, byte_size(Number), Length]
+               ),
+    'undefined'.
+
+-spec should_correct_shortdial(kz_term:ne_binary()) -> {boolean(), pos_integer()}.
+should_correct_shortdial(Number) ->
+    should_correct_shortdial(Number
+                            ,kapps_config:get_integer(?SS_CONFIG_CAT
+                                                     ,<<"min_shortdial_destination">>
+                                                     )
+                            ).
+
+-spec should_correct_shortdial(kz_term:ne_binary(), pos_integer()) -> {boolean(), pos_integer()}.
+should_correct_shortdial(_Number, 'undefined') ->
+    {'true', 0};
+should_correct_shortdial(Number, Length) ->
+    %% The length is also included on the response just to avoid having to read this configuration
+    %% parameter again when printing the debug line. So it is only useful for the `{false, N}' branch.
+    {byte_size(Number) =:= Length, Length}.
+
+-spec do_correct_shortdial(kz_term:ne_binary(), kz_term:ne_binary(), non_neg_integer()) -> kz_term:api_ne_binary().
+do_correct_shortdial(Number, CIDNum, Length) ->
     Correction = kz_binary:truncate_right(CIDNum, Length),
     CorrectedNumber = knm_converters:normalize(<<Correction/binary, Number/binary>>),
-    lager:debug("corrected shortdial ~s via CID ~s to ~s"
-               ,[Number, CIDNum, CorrectedNumber]
-               ),
-    case should_deny_reclassified_number(CorrectedNumber, DeniedCallRestrictions) of
-        'true' ->
-            lager:info("unable to correct shortdial ~s via CID ~s due to a call restriction"
-                      ,[Number, CIDNum]
-                      ),
-            'undefined';
-        'false' ->
-            CorrectedNumber
-    end.
+    lager:debug("corrected shortdial ~s via CID ~s to ~s", [Number, CIDNum, CorrectedNumber]),
+    CorrectedNumber.
 
--spec should_deny_reclassified_number(kz_term:ne_binary(), kz_json:object()) -> boolean().
-should_deny_reclassified_number(CorrectedNumber, DeniedCallRestrictions) ->
+-spec maybe_deny_reclassified_number(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:objects()) -> kz_term:api_binary().
```

Comment:
Spec is wrong, `DeniedCallRestrictions` is a `kz_json:object()` no? At least, previous calls have that var tagged as such.

Reply:
Return should be `kz_term:api_ne_binary()` as well.